apps/hash-agents/docker/Dockerfile

```diff
@@ -22,28 +22,42 @@ RUN yarn install --frozen-lockfile --prefer-offline && \
 COPY --from=base /app/out/full/ .
 
 
-FROM installer AS builder
+FROM python:3.11-alpine as base-env
 
-WORKDIR /usr/local/src/apps/hash-agents
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    WORKDIR="/usr/local/src/hash/apps/hash-agents"
+ENV VENV_DIR="$WORKDIR/.venv"
+ENV PATH="$POETRY_HOME/bin:$VENV_DIR/bin:$PATH"
 
-RUN yarn build
+WORKDIR $WORKDIR
 
 
-FROM python:3.11-alpine as runner
+FROM base-env AS builder
 
-RUN apk add --no-cache g++ musl-dev python3-dev libffi-dev make curl
+COPY --from=installer $WORKDIR/poetry.lock $WORKDIR/pyproject.toml ./
 
-WORKDIR /usr/local/src/apps/hash-agents
+RUN apk add --no-cache --virtual .build-deps g++ musl-dev libffi-dev make curl && \
+    curl -sSL https://install.python-poetry.org | python && \
+    poetry install --no-root --only main && \
+    poetry add 'gunicorn[gevent]' && \
+    apk del .build-deps
 
-COPY --from=builder /usr/local/src/apps/hash-agents/requirements.txt ./requirements.txt
-RUN --mount=type=cache,target=/root/.cache/pip \
-    pip3 install -r requirements.txt
 
-COPY --from=builder /usr/local/src/apps/hash-agents .
+FROM base-env AS runner
+
+WORKDIR $WORKDIR
+
+COPY --from=builder $VENV_DIR $VENV_DIR
```

Comment:
I typically go for taking the same build image in the runner, otherwise you might get some fun runtime exceptions.

Reply:
I know, Python images tend to get huge, just as a word of caution. I had done this myself plenty of times, just to revert it because there was one thing that kept breaking because of it.